osu.Game/Screens/Select/FooterButton.cs

```diff
@@ -6,183 +6,234 @@
-
-        public Color4 DeselectedColour
+        protected Colour4 ButtonColour
         {
-            get => deselectedColour;
             set
             {
-                deselectedColour = value;
-                if (light.Colour != SelectedColour)
-                    light.Colour = value;
+                boxColour.Colour = value;
+                sprite.Colour = value;
             }
         }
 
-        private Color4 selectedColour;
-
-        public Color4 SelectedColour
+        protected IconUsage IconUsageBox
         {
-            get => selectedColour;
-            set
-            {
-                selectedColour = value;
-                box.Colour = selectedColour;
-            }
+            set => sprite.Icon = value;
         }
 
         protected FillFlowContainer ButtonContentContainer;
         protected readonly Container TextContainer;
-        protected readonly SpriteText SpriteText;
-        private readonly Box box;
-        private readonly Box light;
+        private readonly SpriteText spriteText;
+        private readonly Box boxColour;
+        private readonly Box flashLayer;
+        private readonly SpriteIcon sprite;
+        private readonly Box backgroundColourBox;
 
-        public FooterButton()
+        protected FooterButton()
         {
+            Anchor = Anchor.TopLeft;
             AutoSizeAxes = Axes.Both;
             Shear = SHEAR;
-            Children = new Drawable[]
+            CornerRadius = outer_corner_radius;
+            Masking = true;
+            Margin = new MarginPadding { Left = 10, Top = -20 };
+            EdgeEffect = new EdgeEffectParameters
             {
-                box = new Box
-                {
-                    RelativeSizeAxes = Axes.Both,
-                    EdgeSmoothness = new Vector2(2, 0),
-                    Colour = Color4.White,
-                    Alpha = 0,
-                },
-                light = new Box
-                {
-                    Height = 4,
-                    EdgeSmoothness = new Vector2(2, 0),
-                    RelativeSizeAxes = Axes.X,
-                },
-                new Container
+                Type = EdgeEffectType.Shadow,
+                Offset = new Vector2(.250f, 2),
+                Colour = Colour4.Black.Opacity(.25f),
+                Radius = outer_corner_radius
+            };
+
+            {
+                Children = new Drawable[]
                 {
-                    AutoSizeAxes = Axes.Both,
-                    Children = new Drawable[]
+                    backgroundColourBox = new Box
+                    {
+                        Colour = backgroundColour,
+                        RelativeSizeAxes = Axes.Both,
+                        Depth = 2,
+                    },
+                    new Box
+                    {
+                        Alpha = 0.0002f,
+                        Height = 120,
+                        Width = 140,
+                    },
```

Comment:
??

Reply:
i had issues with keeping the container sized appropriately whilst allowing for the expansion with mods (which was a misguided attempt to preserve the previous functionality, my reasoning for leaving it in) will be removed with the rest